http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutes.scala

```diff
@@ -162,11 +164,19 @@ object LeoRoutes {
         )
     }
 
-  def validateClusterNameDirective(clusterNameString: String): Directive1[RuntimeName] =
+  def validateKubernetesName[A](nameString: String, apply: String => A): Directive1[A] =
     Directive { inner =>
-      validateClusterName(clusterNameString) match {
+      KubernetesName.withValidation(nameString, apply) match {
         case Left(e)  => failWith(e)
         case Right(c) => inner(Tuple1(c))
       }
     }
+
```

Comment:
yeah, I'll change this back. Originally I wanted to have it so that all routes validating 'leo entities' the same way, but kubernetes names have a different regex they need to follow